include/maidsafe/crux/detail/multiplexer.hpp

```diff
@@ -59,9 +63,11 @@ class multiplexer : public std::enable_shared_from_this<multiplexer>
     void add(socket_base *);
     void remove(socket_base *);
 
-    template <typename SocketType,
+    template <typename AcceptorType,
+              typename SocketType,
               typename AcceptHandler>
-    void async_accept(SocketType&,
+    void async_accept(AcceptorType*,
```

Comment:
Consider passing `AcceptorType` as reference, just like `SocketType`, which is also stored as a pointer later on. The reason I used pass-by-reference for `SocketType` was to avoid checks for `nullptr`.